lib/index.js

```diff
@@ -1723,38 +1749,33 @@ exposeMethod('pluck');
  */
 
 Stream.prototype.pickBy = function (f) {
-
-    return this.consume(function (err, x, push, next) {
+    return this.map(objectOnly(function (x) {
         var out = {};
-        if (err) {
-            push(err);
-            next();
+        var obj = x;  // variable used to transverse prototype chain
+        function testAndAdd (prop) {
+            if (!(prop in out) && f(prop, x[prop])) {
```

Comment:
I added the test, but reverted my fix when I realized that it didn't originally fail.  The reason why it doesn't fail is that the property may get tested twice, but always with the value relative to the object and not the prototype.
This is because testAndAdd is referrencing from the original object, `obj2` here, rather than the prototype object, `obj1`.  So `v` would always be `2`.

I'm going to go ahead and prevent it from testing a property multiple times.